crypten/mpc/mpc.py

```diff
@@ -218,6 +233,121 @@ def to(self, *args, **kwargs):
             self.share = share
             return self
 
+    def add(self, y):
+        return self.clone().add_(y)
+
+    def add_(self, y):
+        """added addition according to MAC"""
+
+        private = isinstance(y, MPCTensor)
+
+        if not private:
+            self._tensor += y
+            scaled_y = self._tensor.encoder.encode(y)
+            if self._mac is not None:
+                self._mac += self.alpha * scaled_y
+
+        else:
+            self._tensor += y._tensor
+            if self._mac is not None:
+                self._mac += y._mac
+
+        return self
+
+    def mul(self, y):
+        if self._mac is None:
+            if isinstance(y, MPCTensor):
+                y = y._tensor
+            self._tensor = self._tensor.mul_(y)
+        return self.clone().mul_(y)
+
+    def mul_(self, y):
+        """added multiplication according to MAC"""
+
+        private = isinstance(y, MPCTensor)
+
+        if not private:
+            self._tensor *= y
+            scaled_y = self._tensor.encoder.encode(y)
+            if self._mac is not None:
+                self._mac *= scaled_y
+        else:
+            if self._mac is None:
+                self._tensor = self._tensor.mul_(y._tensor)
+            else:
+                raise RuntimeError(...)
```

Comment:
The `...` was a placeholder. This should contain a message about the error. Also looking back, this should be a `NotImplementedError` as it was in the previous PR.